src/TableFunctions/TableFunctionAzureBlobStorage.cpp

```diff
@@ -190,9 +184,91 @@ void TableFunctionAzureBlobStorage::parseArguments(const ASTPtr & ast_function,
 
     auto & args = args_func.at(0)->children;
 
-    configuration = parseArgumentsImpl(args, context);
+    parseArgumentsImpl(args, context);
 }
 
+
+void TableFunctionAzureBlobStorage::addColumnsStructureToArguments(ASTs & args, const String & structure, const ContextPtr & context)
+{
+    if (tryGetNamedCollectionWithOverrides(args, context))
+    {
+        /// In case of named collection, just add key-value pair "structure='...'"
+        /// at the end of arguments to override existed structure.
+        ASTs equal_func_args = {std::make_shared<ASTIdentifier>("structure"), std::make_shared<ASTLiteral>(structure)};
+        auto equal_func = makeASTFunction("equals", std::move(equal_func_args));
+        args.push_back(equal_func);
+    }
+    else
+    {
+        if (args.size() < 3 || args.size() > 8)
+            throw Exception(ErrorCodes::NUMBER_OF_ARGUMENTS_DOESNT_MATCH,
+                            "Storage Azure requires 3 to 7 arguments: "
+                            "AzureBlobStorage(connection_string|storage_account_url, container_name, blobpath, [account_name, account_key, format, compression, structure])");
+
+        auto structure_literal = std::make_shared<ASTLiteral>(structure);
+
+        auto is_format_arg
+            = [](const std::string & s) -> bool { return s == "auto" || FormatFactory::instance().getAllFormats().contains(s); };
+
+
+        if (args.size() == 3)
```

Comment:
It would be good to add a comment with possible signatures before each case like it's done in s3. It will make code more readable and easy to understand

Reply:
addressed it